### ISSUE TITLE
Consider: refactor `types.Destination` as embedded type

### ISSUE THREAD

**Issue by NiloCK** [statechannels/go-nitro#49]
https://github.com/statechannels/go-nitro/pull/42/commits/8c473b3218f6bab3219d6b74cb4e940f241901e5

`types.Destination` is motivated by providing a semantically meaningful narrower subtype to geth's `common.Hash`, which itself is (in essence) a decorated `[32]byte`. Originally, it was an alias to `common.Hash`, but this didn't allow appending the helper functions `ToAddress` and `IsAddress`.

As implemented (a defined type `type Destination common.Hash`) it is necessary to explicitly define forwarding functions (`String`, `Bytes`) that exist on `common.Hash`.

A third option is as a defined type with `common.Hash` as an [embedded type](https://golang.org/doc/effective_go#embedding).

```golang
type Bytes32 = common.Hash


// previous - func (d Destination) destinationHelper(){} not allowed
type Destination = Bytes32

// current - requires reimplementation of methods on Bytes32
type Destination Bytes32

// to investigate - what annoyances lie down this path?
type Destination struct {
  Bytes32
}
```

**Comment by NiloCK** [comment]
I tried this out on branch `embedded-destination-type` - 42ae59d.

**Pro**: forwarding methods are removed from `Destination`, but still available via the embedded `Bytes32` struct
**Con**: constructing **nested** literals becomes still-more verbose, because the linter wants the field `Bytes32` to be labelled. See, for example, `outcome_test.go` in the linked commit.
**Blocker**: the current code fails `go test ./...` because of the abi unmarshaling:

```text
2021/11/16 11:21:29 error unmarshallingjson: cannot unmarshal non-string into Go struct field Allocation.Allocations.Destination of type common.Hash
```

My initial fiddling with `allocationsTy` and `rawAllocationsType` were not successful. @geoknee: you had more success with me in the marshal / unmarshal department. If there is a quick fix that you see we may prefer to merge this. If not, I think the current implementation is not so awful.

**Comment by NiloCK** [comment]
Closed (or deferred) via #62
